packages/loader/container-loader/src/container.ts

```diff
@@ -528,6 +531,20 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         this.removeAllListeners();
     }
 
+    public closeAndGetPendingLocalState(): string {
+        this._deltaManager.close();
```

Comment:
I'd not bake closing into it. I can totally see host calling this API time to time to save container state, but keeping container alive (not to lose user state if app crashes or something like that). Unless we have to to have it for correctness (which I think is not the case - if it is, it's interesting why and might point to bigger problem with our design).